x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/get_context_function_request_if_needed.ts

```diff
@@ -22,11 +22,10 @@ export function getContextFunctionRequestIfNeeded(
     .slice(indexOfLastUserMessage)
     .some((message) => message.message.name === CONTEXT_FUNCTION_NAME);
 
-  if (hasContextSinceLastUserMessage) {
+  const isLastMessageFunctionRequest = !!last(messages)?.message.function_call?.name;
+  if (hasContextSinceLastUserMessage || isLastMessageFunctionRequest) {
```

Comment:
@dgieselaar Does this change look ok to you? Before we'd inject the context request immediately after other function requests effectively overwriting them. This changes will stop us doing that but I'm not sure if it causes problems to context function request injection in some cases